src/index.js

```diff
@@ -197,25 +197,31 @@ class DropdownTreeSelect extends Component {
   }
 
   onKeyboardKeyDown = e => {
-    const { readOnly } = this.props
+    const { readOnly, simpleSelect } = this.props
     const { showDropdown, tags, searchModeOn, currentFocus } = this.state
+    const tm = this.treeManager
+    const tree = searchModeOn ? tm.matchTree : tm.tree
 
     if (!showDropdown && (keyboardNavigation.isValidKey(e.key, false) || /^\w$/i.test(e.key))) {
       // Triggers open of dropdown and retriggers event
       e.persist()
       this.handleClick(null, () => this.onKeyboardKeyDown(e))
       if (/\w/i.test(e.key)) return
     } else if (showDropdown && keyboardNavigation.isValidKey(e.key, true)) {
-      const tm = this.treeManager
-      const tree = searchModeOn ? tm.matchTree : tm.tree
       const newFocus = tm.handleNavigationKey(currentFocus, tree, e.key, readOnly, !searchModeOn, this.onCheckboxChange, this.onNodeToggle)
       if (newFocus !== currentFocus) {
         this.setState({ currentFocus: newFocus })
       }
     } else if (showDropdown && ['Escape', 'Tab'].indexOf(e.key) > -1) {
-      // Triggers close
-      this.keepDropdownActive = false
-      this.handleClick()
+      if (simpleSelect && tree.has(currentFocus)) {
+        this.onCheckboxChange(currentFocus, true)
+      } else {
+        // Triggers close
+        setTimeout(() => {
```

Comment:
`setTimeout`? I haven't looked at the code too deeply but marking this to revisit later.

Reply:
Had issues in firefox with clicking a seach input and shift-tab did not focus the last tag. Worked in chrome but required setTimeout with firefox. I'll have another look later.

Reply:
### Focus click and shift-tab behaviour
✔️ **Chrome no timout**

![chromeNoTimeout](https://user-images.githubusercontent.com/17863113/55916277-38828c80-5bec-11e9-87e2-aa3085f5651f.gif)

⛔️  **Firefox no timout**

![ffNoTimeout](https://user-images.githubusercontent.com/17863113/55916283-3caeaa00-5bec-11e9-8ff2-4f828b2b3782.gif)

✔️ **Firefox with timeout**

![ffSetTimeout](https://user-images.githubusercontent.com/17863113/55916286-3f110400-5bec-11e9-8228-eb8db681a046.gif)

Reply:
Thanks for the GIFs. Am I not able to focus properly or both the cases with FF look the same?

Reply:
I focus with a mouse click and then shift-tab. Chrome and ff with setTimeout focuses on the tag. Ff without focuses on the hierarchial-checkbox.